modules/smartxBidAdapter.js

```diff
@@ -0,0 +1,401 @@
+      }
+      if (!utils.getBidIdParameter('outstream_function', bid.params)) {
+        utils.logMessage(BIDDER_CODE + ': outstream_function parameter is not defined. The default outstream renderer will be injected in the header. You can override the default SmartX outstream rendering by defining your own Outstream function using field outstream_function.');
+        return true;
+      }
+    }
+
+    return true;
+  },
+  /**
+   * Make a server request from the list of BidRequests.
+   * from Prebid.js: buildRequests - Takes an array of valid bid requests, all of which are guaranteed to have passed the isBidRequestValid() test.
+   *
+   * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (bidRequests, bidderRequest) {
+    const page = bidderRequest.refererInfo.referer;
+    const isPageSecure = !!page.match(/^https:/)
+
+    const smartxRequests = bidRequests.map(function (bid) {
+      const tagId = utils.getBidIdParameter('tagId', bid.params);
+      const publisherId = utils.getBidIdParameter('publisherId', bid.params);
+      const bidfloor = utils.getBidIdParameter('bidfloor', bid.params);
+      const bidfloorcur = utils.getBidIdParameter('bidfloorcur', bid.params);
+      const siteId = utils.getBidIdParameter('siteId', bid.params);
+      const domain = utils.getBidIdParameter('domain', bid.params);
+      const cat = utils.getBidIdParameter('cat', bid.params);
+      let pubcid = null;
+      const playerSize = utils.deepAccess(bid, 'mediaTypes.video.playerSize');
+      const contentWidth = playerSize[0][0];
+      const contentHeight = playerSize[0][1];
+      const secure = +(isPageSecure || (utils.getBidIdParameter('secure', bid.params) ? 1 : 0));
+      const ext = {
+        sdk_name: 'Prebid 1+'
+      };
+      const mimes = utils.getBidIdParameter('mimes', bid.params) || ['application/javascript', 'video/mp4', 'video/webm'];
+      const linearity = utils.getBidIdParameter('linearity', bid.params) || 1;
+      const minduration = utils.getBidIdParameter('minduration', bid.params) || 0;
+      const maxduration = utils.getBidIdParameter('maxduration', bid.params) || 500;
+      const startdelay = utils.getBidIdParameter('startdelay', bid.params) || 0;
+      const minbitrate = utils.getBidIdParameter('minbitrate', bid.params) || 0;
+      const maxbitrate = utils.getBidIdParameter('maxbitrate', bid.params) || 3500;
+      const delivery = utils.getBidIdParameter('delivery', bid.params) || [2];
+      const pos = utils.getBidIdParameter('pos', bid.params) || 1;
+      const api = utils.getBidIdParameter('api', bid.params) || [2];
+      const protocols = utils.getBidIdParameter('protocols', bid.params) || [2, 3, 5, 6];
+      var contextcustom = utils.deepAccess(bid, 'mediaTypes.video.context');
+      var placement = 1;
+
+      if (contextcustom == 'outstream') {
```

Comment:
Triple equals.

The context in the `bid.params` is not taken into account. See my comment below. Maybe the `context` in the params is redundant and can be removed.

Reply:
done